src/test/java/htsjdk/samtools/util/IOUtilTest.java

```diff
@@ -425,4 +427,64 @@ public void testUnrollPaths(final String pathUri, final String[] extensions, fin
     }
+
+    @DataProvider(name = "blockCompressedExtensionExtensionStrings")
+    public static Object[][] createBlockCompressedExtensionStrings() {
+        return new Object[][] {
+                { "testzip.gz", true },
+                { "test.gzip", true },
+                { "test.bgz", true },
+                { "test.bgzf", true },
+                { "test.bzip2", false }
+        };
+    }
+
+    @Test(dataProvider = "blockCompressedExtensionExtensionStrings")
+    public void testBlockCompressionExtensionString(final String testString, final boolean expected) {
+        Assert.assertEquals(IOUtil.hasBlockCompressedExtension(testString), expected);
+    }
+
+    @Test(dataProvider = "blockCompressedExtensionExtensionStrings")
+    public void testBlockCompressionExtensionFile(final String testString, final boolean expected) {
+        Assert.assertEquals(IOUtil.hasBlockCompressedExtension(new File(testString)), expected);
+    }
+
+    @DataProvider(name = "blockCompressedExtensionExtensionURIStrings")
+    public static Object[][] createBlockCompressedExtensionURIs() {
+        return new Object[][]{
+                {"testzip.gz", true},
+                {"test.gzip", true},
+                {"test.bgz", true},
+                {"test.bgzf", true},
+                {"test", false},
+                {"test.bzip2", false},
+
+                {"https://www.googleapis.com/download/storage/v1/b/deflaux-public-test/o/NA12877.vcf.gz", true},
+                {"https://www.googleapis.com/download/storage/v1/b/deflaux-public-test/o/NA12877.vcf.gzip", true},
+                {"https://www.googleapis.com/download/storage/v1/b/deflaux-public-test/o/NA12877.vcf.bgz", true},
+                {"https://www.googleapis.com/download/storage/v1/b/deflaux-public-test/o/NA12877.vcf.bgzf", true},
+                {"https://www.googleapis.com/download/storage/v1/b/deflaux-public-test/o/NA12877.vcf.bzip2", false},
+                {"https://www.googleapis.com/download/storage/v1/b/deflaux-public-test/o/NA12877", false},
+
+                {"https://www.googleapis.com/download/storage/v1/b/deflaux-public-test/o/NA12877.vcf.gz?alt=media", true},
+                {"https://www.googleapis.com/download/storage/v1/b/deflaux-public-test/o/NA12877.vcf.gzip?alt=media", true},
+                {"https://www.googleapis.com/download/storage/v1/b/deflaux-public-test/o/NA12877.vcf.bgz?alt=media", true},
+                {"https://www.googleapis.com/download/storage/v1/b/deflaux-public-test/o/NA12877.vcf.bgzf?alt=media", true},
+                {"https://www.googleapis.com/download/storage/v1/b/deflaux-public-test/o/NA12877.vcf.bzip2?alt=media", false},
+
+                {"ftp://ftp.broadinstitute.org/distribution/igv/TEST/cpgIslands.hg18.gz", true},
+                {"ftp://ftp.broadinstitute.org/distribution/igv/TEST/cpgIslands.hg18.bed", false}
+        };
+    }
+
+    @Test(dataProvider = "blockCompressedExtensionExtensionURIStrings")

```

Comment:
What do you mean with path + url parameters? It is already covered (e.g., `https://www.googleapis.com/download/storage/v1/b/deflaux-public-test/o/NA12877.vcf.gz?alt=media`.

I will prepare a test for `java.nio.Path` using Jimfs, but let me know if it is what you want here....

Reply:
What I was thinking was creating a path from the string with url parameters, but that doesn't actually make sense since we don't have an http filesystem.  Sorry, no action needed here, I wasn't thinking it through.